vertexai/src/ModelContent.cs

```diff
@@ -311,21 +311,36 @@ internal static ModelContent FromJson(Dictionary<string, object> jsonDict) {
       jsonDict.ParseObjectList("parts", PartFromJson, JsonParseOptions.ThrowEverything));
   }
 
-  private static FunctionCallPart FunctionCallPartFromJson(Dictionary<string, object> jsonDict) {
-    return new FunctionCallPart(
-      jsonDict.ParseValue<string>("name", JsonParseOptions.ThrowEverything),
-      jsonDict.ParseValue<Dictionary<string, object>>("args", JsonParseOptions.ThrowEverything));
+  private static InlineDataPart InlineDataPartFromJson(Dictionary<string, object> jsonDict) {
```

Comment:
Why this is modifying the FunctionCallPart? I think we would need both.

Reply:
I moved it below, into an internal ModelContentJsonParsers class, so that LiveContentResponse could reuse the parsing logic.